### PR TITLE
[SQL] Add INSERT tests to JdbcDriverTest

### PR DESCRIPTION
**Pull request by akedin**
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

**Comment by akedin** [comment]
R: @apilloud @kennknowles @amaliujia

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/parser/TestTableProvider.java [hunk 1]
```diff
@@ -17,38 +17,165 @@
...
+/**
+ * Test in-memory table provider for use in tests.
+ *
+ * <p>Keeps global state and tracks class instances. Works only in DirectRunner.
+ *
+ * <p>Similar to {@link MockedBoundedTable} with more concurrency support.
+ */
 public class TestTableProvider implements TableProvider {
-  private Map<String, Table> tables = new HashMap<>();
+  static final Map<Long, Map<String, TableWithRows>> GLOBAL_TABLES = new ConcurrentHashMap<>();
```
Comment (amaliujia) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/parser/TestTableProvider.java]: Could you elaborate why convert tables to static and change `BeamSqlTable` to `synchronized`?

It seems like it is changed for multi-threading?

Comment (akedin) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/parser/TestTableProvider.java]: This is the same idea as in EnumerableConverter. We keep accumulating data in a global state. The state is keyed by the instance ID of the table provider, which creates tables. Few queries can run in parallel against different tables, so we need something like this

**Comment by akedin** [comment]
ping

**Comment by kennknowles** [comment]
LGTM

**Pull request merged**